addons/source-python/packages/source-python/core/__init__.py

```diff
@@ -101,6 +107,36 @@ def _unload_instance(self):
             'have its own implementation of an _unload_instance method.')
 
 
+class WeakAutoUnload(AutoUnload):
+    """Subclass of AutoUnload used to unload specific instances when all
+    references to the instance are deleted.
+    """
+
+    def new(cls, *args, **kwargs):
+        self = super().__new__(cls)
+        self.___unloaded = False
```

Comment:
Somewhere we defined that we don't want to use super-private attributes. That's why we never use 2 underscores in SP. It seems like that point is missing in the wiki.